### PR TITLE
wasm: Force stop iteration after local response is sent

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: mathetake <takeshi@tetrate.io>

resolves https://github.com/envoyproxy/envoy/issues/13857

ref: 
-https://github.com/proxy-wasm/proxy-wasm-rust-sdk/issues/44
-https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/88
-https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/93

### CODE REVIEW

**Review by PiotrSikora — APPROVED** [review]
LGTM once the repository updated to point to the merged commit.

**Review by PiotrSikora — APPROVED** [review]

**Review by PiotrSikora — CHANGES_REQUESTED** [review]
Something is broken.

/wait

**Review by PiotrSikora — DISMISSED** [review]

File: bazel/repository_locations.bzl
Comment (PiotrSikora) [inline bazel/repository_locations.bzl:837]: You need to update `release_date` below to `2020-11-10`.

**Review by PiotrSikora — APPROVED** [review]

File: test/extensions/filters/http/wasm/wasm_filter_test.cc
Comment (antoniovicente) [inline test/extensions/filters/http/wasm/wasm_filter_test.cc:1440]: Given that the filter code is in a different repo, does it make sense to have assertions like this one about the return values of decodeHeaders in the main repo?

Comment (PiotrSikora) [reply test/extensions/filters/http/wasm/wasm_filter_test.cc:1440]: The test filter is in this repo, this is the test case for which the behavior changed:
https://github.com/envoyproxy/envoy/blob/1d6d68bd659a78ad53b46799efa2f810c3eb2dd7/test/extensions/filters/http/wasm/test_data/test_cpp.cc#L750-L754

Comment (antoniovicente) [reply test/extensions/filters/http/wasm/wasm_filter_test.cc:1440]: So the coding error was in a wasm filter implementation, not in the wasm framework itself?

**Review by lizan — APPROVED** [review]

**Pull request merged**
